InventoryBuilder.py

```diff
@@ -118,9 +119,35 @@ def query_inventory_permanent(self):
             # initialize empty inventory per iteration
             self.iterated_inventory[str(self.iteration)] = dict()
             logger.info(f'real run {self.iteration}')
+            threads = list()
             for vrops in self.vrops_list:
-                if not self.query_vrops(vrops):
-                    logger.warning(f'retrying connection to {vrops} in next iteration {self.iteration + 1}')
+                vrops_short_name = vrops.split('.')[0]
+                thread = Thread(target=self.query_vrops, args=(vrops, vrops_short_name))
+                thread.start()
+                threads.append((thread, vrops))
+
+            timeout = self.timeout
+            start_time = time.time()
+            current_time = start_time
+            joined_threads = dict()
+            while current_time <= (start_time + timeout):
+                for t in threads:
+                    if not t[0].is_alive():
+                        t[0].join()
+                        if t[0] not in joined_threads:
+                            joined_threads.setdefault(t[1], round(time.time() - start_time))
+                if len(joined_threads.keys()) >= len(threads):
+                    break
+                time.sleep(1)
+                current_time = time.time()
+            else:
+                still_running = [t for t in threads if t[0].is_alive()]
+                for running_thread in still_running:
+                    logger.info(f"Timeout {timeout}s reached for fetching {running_thread[1]}")
```

Comment:
Alright, so use the timeout parameter to close the thread if it gets stuck forever?